src/hera/v1/task.py

```diff
@@ -163,6 +164,28 @@ def __rshift__(self, other: 'Task') -> 'Task':
         """
         return self.next(other)
 
+    def when(self, other: 'Task', operator: 'Operator', value: str):
+        """Sets this task as a dependency of the other passed task if the condition match.
+
+        Parameters
+        ----------
+        other: Task
+            The other task to set a dependency for. The new dependency of the task is this task.
+
+        Returns
+        -------
+        Task
+            The other task that was specified.
+
+        Examples
+        --------
+        t1, t2, t3 = Task('t1'), Task('t2'), Task('t3')
+        t2.when(t1, Operator.equals, "t2")
+        t3.when(t1, Operator.equals, "t3")
+        """
+        self.argo_task.when = f'{{{{tasks.{other.name}.outputs.result}}}} {operator} {value}'
```

Comment:
If we switch the `Operator` to an `Enum`, then this becomes: 
```suggestion
        self.argo_task.when = f'{{{{tasks.{other.name}.outputs.result}}}} {operator.value} {value}'
```